.github/workflows/docs.yaml

```diff
@@ -0,0 +1,62 @@
+# Sample workflow for building and deploying a mdBook site to GitHub Pages
+#
+# To get started with mdBook see: https://rust-lang.github.io/mdBook/index.html
+#
+name: Deploy mdBook site to Pages
+
+on:
+  # Runs on only core version tags
+  push:
+    tags:
+      - 'v*'
```

Comment:
Does this imply that the docbook has an ability to render different version-time views of the docs? Like a `release-0.2` view vs `release-0.3` view? If not, we may need to make sure that we are only releasing doc updates from a specific branch.

Reply:
We can turn off automatic build and rely only on manual action triggers, which low us to select branch from which the page is built.

Only the latest build is displayed, mdbook does not support versioning. I think I can add additional metadata to the page that will indicate the time/version/commit sha from which the documentation was built.

Reply:
Done. If built from tagged version, it will display Version now. Otherwise, it will display only commit sha.

Reply:
<img width="812" alt="image" src="https://github.com/user-attachments/assets/b661a3d2-7a8f-4364-b44f-0cc754f47f26" />